internal/server/manager.go

```diff
@@ -37,18 +38,32 @@ type ClusterInfo struct {
 	FailureMessage *string           `json:"failureMessage"`
 }
 
-type instance struct {
-	client     client.Client
-	scheme     *runtime.Scheme
-	clusterMux sync.Mutex // use a Mutex to update managed Clusters
+type ClusterProfileStatus struct {
+	Name        *string                         `json:"name"`
+	Namespace   *string                         `json:"namespace"`
+	ClusterName *string                         `json:"clusterName"`
+	Summary     []configv1alpha1.FeatureSummary `json:"summary"`
+}
 
-	sveltosClusters map[corev1.ObjectReference]ClusterInfo
-	capiClusters    map[corev1.ObjectReference]ClusterInfo
+type instance struct {
+	client             client.Client
+	scheme             *runtime.Scheme
+	clusterMux         sync.Mutex // use a Mutex to update managed Clusters
+	clusterStatusesMux sync.Mutex // mutex to update cached ClusterSummary instances
+
+	sveltosClusters        map[corev1.ObjectReference]ClusterInfo
+	capiClusters           map[corev1.ObjectReference]ClusterInfo
+	clusterProfileStatuses map[corev1.ObjectReference]ClusterProfileStatus
```

Comment:
nit: feel free to ignore this. I got confused because it  is called clusterProfilesStatuses while in reality the key is a ClusterSummary.

I suggest to call this clusterSummaryStatus or better clusterSummaryReport